### PR TITLE
fix: simplify logic of update env file, and fix bug

### PR DESCRIPTION
**Pull request by Ace-Tang**
since updateContainerEnv logic is too complicated,
content in /etc/profile.d/pouchenv.sh should same with config env,
cover env file with merged env.
fix check same value error:
'msg="env exists and the value is not same, key=spas_accessKey, old
value="4f032bc4b0db47db9baa95fcd4c7b3ad", new value=4f032bc4b0db47db9baa95fcd4c7b3ad"'

Signed-off-by: Ace-Tang <aceapril@126.com>

### Ⅰ. Describe what this PR did

fix failed to update env file with old logic.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

add rich container update env ci test.

### CODE REVIEW

File: daemon/mgr/container_envs.go [hunk 1]
```diff
@@ -24,110 +20,40 @@ import (
...
-		return nil
-	}
-	if _, err := os.Stat(envShPath); err != nil {
-		logrus.Warnf("failed to stat container's env file /etc/profile.d/pouchenv.sh: %v", err)
+func updateContainerEnv(env []string, baseFs string) error {
+	// if target env directory is not exist, return
+	// generally speaking, we should check env file exist to decide
+	// whether to update env file, but in case of missing something,
+	// we check directory existence here to make a decision.
+	if _, err := os.Stat(filepath.Join(baseFs, filepath.Dir(pouchEnvFile))); err != nil {
```
Comment (HusterWan) [inline daemon/mgr/container_envs.go]: why we not check the file existence, but check the dir?

File: daemon/mgr/container_envs.go [hunk 2]
```diff
@@ -24,110 +20,40 @@ import (
...
+			s = strings.Replace(s, "\"", "\\\"", -1)
+			s = strings.Replace(s, "$", "\\$", -1)
+			if parts[0] == "PATH" {
+				s = parts[1] + ":$PATH"
+			}
+			str += fmt.Sprintf("export %s=\"%s\"\n", parts[0], s)
 		}
 	}
-	return localEnv
+	ioutil.WriteFile(filepath.Join(baseFs, pouchEnvFile), []byte(str), 0755)
```
Comment (HusterWan) [inline daemon/mgr/container_envs.go]: OK, i got you, we will just write the env to the file, so we just need check the existence of Dir

File: daemon/mgr/container.go [hunk 3]
```diff
@@ -999,16 +1006,12 @@ func (mgr *ContainerManager) Update(ctx context.Context, name string, config *ty
...
-	if err != nil {
-		return err
-	}
-	c.Config.Env = newEnvSlice
-
 	if mgr.containerPlugin != nil {
-		if err = mgr.containerPlugin.PostUpdate(c.BaseFS, c.Config.Env); err != nil {
-			return err
+		// post update env file
+		if len(config.Env) > 0 {
```
Comment (HusterWan) [inline daemon/mgr/container.go]: how about `if len(config.Env) > 0 && err := mgr.containerPlugin.PostUpdate(c.BaseFS, c.Config.Env); err != nil` ?

Comment (Ace-Tang) [reply daemon/mgr/container.go]: Too long, what do you think?

Comment (Ace-Tang) [reply daemon/mgr/container.go]: Look like conbine `mgr.containerPlugin != nil && len(config.Env) > 0` better?

Comment (HusterWan) [reply daemon/mgr/container.go]: can not agree no more

Comment (Ace-Tang) [reply daemon/mgr/container.go]: Updated.

**Comment by HusterWan** [comment]
lgtm

File: daemon/mgr/container.go [hunk 4]
```diff
@@ -989,6 +989,13 @@ func (mgr *ContainerManager) Update(ctx context.Context, name string, config *ty
 	}
 	c.Unlock()
 
+	// Update Env
```
Comment (HusterWan) [inline daemon/mgr/container.go]: update container parameters, should we add `Lock` here?

Comment (Ace-Tang) [reply daemon/mgr/container.go]: This should be disscuss in another pr, we are not determine to modify origin logic here, or you can raise a pr if  you think if lock should be added.

**Pull request merged**